packages/sdk-rtl/src/lookerSDKError.ts

```diff
@@ -0,0 +1,191 @@
+  documentation_url?: string | null
+}
+
+interface ILookerSDKErrorConstructor {
+  new (
+    ...args: AugmentErrorOptions<
+      ConstructorParameters<ErrorConstructor>,
+      ILookerSDKErrorOptions
+    >
+  ): LookerSDKError
+  (
+    ...args: AugmentErrorOptions<
+      Parameters<ErrorConstructor>,
+      ILookerSDKErrorOptions
+    >
+  ): LookerSDKError
+}
+
+// The subclass and function expression's name should match, so that stack traces look clean.
+// We bind it to a local identifier for clarity, and to perform a type assertion.
+export interface LookerSDKError extends Error {
+  errors?: IErrorDetail[]
+  documentation_url?: string | null
+}
+
+export const LookerSDKError: ILookerSDKErrorConstructor =
+  /* #__PURE__*/ (() => {
+    'use strict'
+    const LookerSDKErrorConstructor = function LookerSDKError(
+      this: LookerSDKError | undefined,
+      ...[
+        message,
+        { errors = [], documentation_url = '', ...errorOptions } = {},
+        ...rest
+      ]: AugmentErrorOptions<
+        ConstructorParameters<ErrorConstructor>,
+        ILookerSDKErrorOptions
+      > &
+        AugmentErrorOptions<
+          Parameters<ErrorConstructor>,
+          ILookerSDKErrorOptions
+        >
+    ) {
+      // The `super()` call. At present, Error() and new Error() are
+      // indistinguishable, but use whatever we were invoked with in case
+      // that ever changes.
+      const error = this
+        ? new Error(
+            message,
+            // we have to suppress a type error here if TypeScript
+            // doesn't know es2022's two-argument Error constructor
+            // @ts-ignore-error
+            errorOptions,
+            ...rest
+          )
+        : Error(
+            message,
+            // @ts-ignore-error
+            errorOptions,
+            ...rest
+          )
+
+      // Object.setPrototypeOf() is necessary when extending built-ins,
+      // since Error.call(this, message, errorOptions, ...rest) doesn't
+      // set up the prototype chain the way it would with a user-defined
+      // class.
+      Object.setPrototypeOf(
+        error,
+        this ? Object.getPrototypeOf(this) : LookerSDKError.prototype
+      )
+
+      // Normally the 'message' property of JavaScript Error objects is non-enumerable
+      // set it to be enumerable for consistency with our non-error responses,
+      // which are implemented as plain JavaScript objects where all
+      // properties are enumerable.
+      Object.defineProperties(error, {
+        message: {
+          enumerable: true,
+        },
+        errors: {
+          value: errors ?? [], // defaults to an empty array if null or undefined
+          writable: true,
+          configurable: true,
+          enumerable: true,
+        },
+        documentation_url: {
+          value: documentation_url ?? '',
+          writable: true,
+          configurable: true,
+          enumerable: true,
+        },
+      })
```

Comment:
* Change enumerability per discussion with @AprilArcus and @jkaster. `Object.keys(someError)` now returns `['message', 'errors', 'documentation_url']`

* Add `null` handling for `errors` and `documentation_url` according to the comment in the previous PR: https://github.com/looker-open-source/sdk-codegen/pull/1134

Reply:
I'd suggest writing this in the simpler style:

```
Object.defineProperty(error, 'message', { enumerable: true })
error.errors = errors
error.documentation_url = documentation_url
```

three lines vs. sixteen.

Reply:
@AprilArcus I had tried this based on the discussion on the previous PR. I have got the error below, so I went with the code above.
```
error TS2339: Property 'errors' does not exist on type 'Error'.
```

Reply:
Do a cast first. `(error as LookerSDKError).errors = errors`